### PR TITLE
ml_compute_instance supports creation_script and startup_script

### PR DESCRIPTION
**Pull request by ms-henglu**
image
image

There's a test failed because of duplicated name in key vault. So I combined tests results from 2 builds.

### CODE REVIEW

File: internal/services/machinelearning/machine_learning_compute_instance_resource.go [hunk 1]
```diff
@@ -247,6 +310,10 @@ func resourceComputeInstanceRead(d *pluginsdk.ResourceData, meta interface{}) er
 			d.Set("authorization_type", props.Properties.ComputeInstanceAuthorizationType)
 			d.Set("ssh", flattenComputeSSHSetting(props.Properties.SSHSettings))
 			d.Set("assign_to_user", flattenComputePersonalComputeInstanceSetting(props.Properties.PersonalComputeInstanceSettings))
+			if props.Properties.SetupScripts != nil && props.Properties.SetupScripts.Scripts != nil {
+				d.Set("startup_script", flattenComputeScript(props.Properties.SetupScripts.Scripts.StartupScript))
+				d.Set("creation_script", flattenComputeScript(props.Properties.SetupScripts.Scripts.StartupScript))
```
Comment (katbyte) [inline internal/services/machinelearning/machine_learning_compute_instance_resource.go:315]: should this be?
```suggestion
				d.Set("creation_script", flattenComputeScript(props.Properties.SetupScripts.Scripts. CreationScript))
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
@katbyte , Hi, thanks for pointing it out, I've fixed it, please take another look, thanks!

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 🚀

**Review by katbyte — CHANGES_REQUESTED** [review]
could we update/fix the tests so they pass when run at the same time please?
```
------ Stdout: -------
=== RUN   TestAccComputeCluster_basic
=== PAUSE TestAccComputeCluster_basic
=== CONT  TestAccComputeCluster_basic
    testcase.go:88: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating Vault: (Name "acctestvault210830113699" / Resource Group "acctestRG-ml-210830113653483699"): keyvault.VaultsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="VaultAlreadyExists" Message="The vault name 'acctestvault210830113699' is already in use. Vault names are globaly unique so it is possible that the name is already taken. If you are sure that the vault name was not taken then it is possible that a vault with the same name was recently deleted but not purged after being placed in a recoverable state. If the vault is in a recoverable state then the vault will need to be purged before reusing the name. For more information on soft delete and purging a vault follow this link https://go.microsoft.com/fwlink/?linkid=2147740."
        
          with azurerm_key_vault.test,
          on terraform_plugin_test.tf line 24, in resource "azurerm_key_vault" "test":
          24: resource "azurerm_key_vault" "test" {
        
--- FAIL: TestAccComputeCluster_basic (164.97s)
FAIL
```

**Comment by ms-henglu** [comment]
@katbyte , Hi, keyvault's name updated.
image

**Comment by katbyte** [comment]
Looks like we have a transitory failure? 
```
------- Stdout: -------
=== RUN   TestAccComputeInstance_complete
=== PAUSE TestAccComputeInstance_complete
=== CONT  TestAccComputeInstance_complete
    testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Subnet: (Name "internal" / Virtual Network Name "acctest21083190" / Resource Group "acctestRG-ml-210831223127158990"): network.SubnetsClient#Delete: Failure sending request: StatusCode=0 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet internal is in use by /subscriptions/9327cac6-f79a-46ec-a3fb-75563f3f7b8e/resourceGroups/f6341291-5815-4ad7-9e7c-25a1d2fd5319-AzureBatch-CloudService/providers/Microsoft.Network/networkInterfaces/|providers|Microsoft.Compute|virtualMachineScaleSets|f6341291-5815-4ad7-9e7c-25a1d2fd5319-AzureBatch-Deployment|virtualMachines|1|networkInterfaces|nic1/ipConfigurations/ip1 and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]
        
--- FAIL: TestAccComputeInstance_complete (1066.33s)
FAIL
```

**Comment by ms-henglu** [comment]
@katbyte , please hold on merging this PR, I got some requests from service team.

**Comment by katbyte** [comment]
@ms-henglu - given this is sitting blocked on the service team iu am going to close it, please reopen it once its ready to go!

**Pull request closed without merging**